tests/commands/timesource.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright 2020 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use, modify,
+# copy, or redistribute it subject to the terms and conditions of the GNU
+# General Public License v.2.  This program is distributed in the hope that it
+# will be useful, but WITHOUT ANY WARRANTY expressed or implied, including the
+# implied warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  Any Red Hat
+# trademarks that are incorporated in the source code or documentation are not
+# subject to the GNU General Public License and may only be used or replicated
+# with the express permission of Red Hat, Inc.
+#
+
+import unittest
+
+from tests.baseclass import CommandTest
+from pykickstart.commands.timesource import F33_TimesourceData
+
+class F33_TestCase(CommandTest):
+    command = "timesource"
+
+    def runTest(self):
+        data1 = F33_TimesourceData()
+        data1.ntp_server="ntp.example.com"
+        data2 = F33_TimesourceData()
+        data2.ntp_server="ntp.example.com"
+        self.assertEqual(data1, data2)
+        self.assertFalse(data1 != data2) # test __ne__ method
+        self.assertNotEqual(data1, None)
+
+        for attr in ['ntp-server', 'ntp-pool']:
+            data1 = F33_TimesourceData()
+            data2 = F33_TimesourceData()
+            setattr(data1, attr, '')
+            setattr(data2, attr, 'test')
+            self.assertNotEqual(data1, data2)
+            self.assertNotEqual(data2, data1)
+            setattr(data1, attr, '')
+            setattr(data2, attr, '')
+
+        # pass
+        self.assert_parse("timesource --ntp-server=ntp.example.com")
```

Comment:
It would be nice to have at least one test with multiple sources.

Reply:
Added.

Reply:
Added.